ding_periodical.module

```diff
@@ -135,12 +135,14 @@ function theme_ding_periodical_issues($variables) {
 
       // Check if the volume is reservable and add reservation button.
       if (!empty($issue_id) && ding_periodical_is_reservable($issues[$key])) {
-        $path = array($issue_id, $volume, $key);
+        $item_id = array($issue_id, $volume, $key);
+        $item_id = array_map('base64_encode', $item_id);
+
         $button = array(
           array(
             '#theme' => 'link',
             '#text' => t('Reserve'),
-            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $path),
+            '#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . base64_encode(implode('$', $item_id)),
```

Comment:
If we `base64_encode()` on line 139 there is no need to do it again here.

This line could should be:

```php
'#path' => 'ting/object/' . $variables['entity']->id . '/reserve/' . implode('/', $item_id),
```